Structural/Decorator.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <string>
+
+// Component
+class IceCream
+{
+public:
+    virtual int kcal() = 0;
+    virtual int price() = 0;
+};
+
+// Concrete Component
+class ClassicIceCream : public IceCream
+{
+public:
+    int kcal()
+    {
+        return 100;
```

Comment:
It would be great if you could define price and calories in class members and not return magic numbers in functions, it would be more readable